R/app_alphaDiv.R

```diff
@@ -5,22 +5,29 @@
 #' @param df data.table with rows corresponding to records, one column indicating the record id, and all other columns corresponding to taxa abundance.
 #' @param recordIdColumn string defining the name of the df column that specifies record ids. Note, all other columns must be numeric and will be treated as abundance values.
 #' @param method string defining the the alpha diversity method. Accepted values are 'shannon','simpson', and 'evenness'
+#' @param naToZero boolean indicating if NAs in numeric columns should be replaced with 0
 #' @param verbose boolean indicating if timed logging is desired
 #' @return data.table with columns recordIdColumn, "alphaDiversity".
 #' @importFrom vegan diversity
 #' @importFrom stringi stri_trans_totitle
 #' @import veupathUtils
 #' @import data.table
 #' @export
-alphaDiv <- function(df, recordIdColumn, method = c('shannon','simpson','evenness'), verbose = c(TRUE, FALSE)) {
+alphaDiv <- function(df, recordIdColumn, method = c('shannon','simpson','evenness'), naToZero = c(FALSE, TRUE), verbose = c(TRUE, FALSE)) {
```

Comment:
i get it probably makes sense in general/ ideally for the default to be false. but its worth noting that at least until we have the data annotated properly, we may have to have the default true here.

Reply:
hmm you're right we'll definitely be needing the arg to be true for a while :) 

When we do have the annotations, i can set the default back to false. I'm hoping it will be helpful for data qa because we'll easily find out which vars should be annotated but aren't!